osmo_jupyter/plot.py

```diff
@@ -0,0 +1,224 @@
+        'titlefont': {'color': color},
+        'tickfont': {'color': color},
+        'tickcolor': color,
+    } if color else {}
+    return {
+        f'yaxis{axis_number}': {
+            'title': title,
+            **color_kwargs,
+            **y_axis_params
+        }
+    }
+
+
+def _add_dicts(dicts):
+    output_dict = {}
+    for _dict in dicts:
+        output_dict.update(_dict)
+    return output_dict
+
+
+def get_rgb_columns(df, pattern, other_columns_to_include=None):
+    ''' Get columns from the input DataFrame corresponding to colors, removing a shared prefix or suffix
+
+    eg.
+    >>> df = pd.DataFrame(..., columns=['r_mean', 'g_mean', 'b_mean', 'Timestamp', 'other', ...])
+    >>> get_rgb_columns(df, '*_mean', ['Timestamp'])
+    <DataFrame with columns ['r', 'g', 'b', 'Timestamp'] >
+
+    Params:
+        df: pandas.DataFrame of source data
+        pattern: string with an asterisk ("*") in place of the color letter, eg. '*_mean'
+        other_columns_to_include: list of column names to be included
+
+    '''
+    other_columns_to_include = other_columns_to_include or []
+
+    color_columns = [
+        pattern.replace('*', color)
+        for color in COLOR_CHANNELS
+    ]
+
+    all_desired_columns = color_columns + other_columns_to_include
+    rename_map = {
+        original_column: new_column
+        for original_column, new_column in zip(color_columns, COLOR_CHANNELS)
+    }
+    return df[all_desired_columns].rename(columns=rename_map)
+
+
+def get_layout_with_annotations(
+    x_axis_title='Time',
+    y_axis_title=None,
+    colors_on_separate_axes=False,
+    events=None,
+    **additional_layout_params
+):
+    ''' Get a go.Layout instance prepared for our standard graphing needs
+
+    Notes re. our standard layout:
+    * The primary Y axis (y0) is reserved for annotations. This is done to guarantee that annotations show
+    * If colors are on separate axes, they will be on the next consecutive axes, otherwise they will be on yaxis 2
+    * You can use additional axes for plotting other things by passing in additional params
+
+    Params:
+        x_axis_title: title to use for X axis, default 'Time'
+        y_axis_title: title to use for Y axis where colors are plotted. Required if colors_on_separate_axes is False.
+            If colors_on_separate_axes is True, it will be used as a suffix for the y axis titles "r", "g", and "b"
+        colors_on_separate_axes: whether colors (r,g and b) should be displayed on separate Y axes
+        events: dictionary of {x_axis_value: "annotation"} which will be used to annotate the chart
+        additional_layout_params: These will be passed to go.Layout. These can, for instance, be the result of
+            axis_kwargs() to add support for graphing more items.
+    '''
+    events = events or {}
+    if not colors_on_separate_axes and not y_axis_title:
+        raise ValueError('If colors_on_separate_axes is False you must provide a y_axis_title')
```

Comment:
because you should always label your axes.
good point though I don't really need to enforce that here; I can just let y_axis title be an empty string by default.

Reply:
yeah seems kind of annoying to enforce that when you might just want to throw a quick graph together

Reply:
removed the requirement.